dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/jobService/PutJob.kt

```diff
@@ -32,24 +32,32 @@ import io.reactivex.disposables.Disposable
 import org.slf4j.LoggerFactory
 import java.util.*
 import java.util.concurrent.TimeUnit
+import java.util.function.Supplier
+import kotlin.reflect.KFunction0
 
 class PutJob(private val putJobData: JobData) : JobService(), PrepStage<JobData>, TransferStage, TeardownStage {
     override fun getDs3Client(): Ds3Client = putJobData.client()
 
     private var job: Ds3ClientHelpers.Job? = null
     private val chunkManagment: ChunkManagment = ChunkManagment()
     private val stats: Stats = Stats(message, putJobData.loggingService(), putJobData.dateTimeUtils())
+
     private companion object {
         private val LOG = LoggerFactory.getLogger(GetJob::class.java)
     }
 
     override fun jobUUID(): UUID? = putJobData.jobId
 
-    override fun finishedCompletable(): Completable {
+    override fun finishedCompletable(cancelled: Supplier<Boolean>): Completable {
+        putJobData.cancelled = cancelled
         return Completable.fromAction {
             val resources = prepare(putJobData)
-            transfer(resources)
-            tearDown()
+            if (cancelled.get() == false) {
```

Comment:
Earlier, you had `!cancelled.get()` instead of this explicit comparison.  Pick on or the other.  And I am curious if the alternative `return@` syntax will work here.